assets/signing/sign-message.bouncycastle.cs

```diff
@@ -0,0 +1,73 @@
+// #########################################################
+// #             WARNING   WARNING   WARNING               #
+// #########################################################
+// #                                                       #
+// # This file is intended for demonstration purposes      #
+// # only.                                                 #
+// #                                                       #
+// # It is the SOLE responsibility of YOU, the programmer  #
+// # to prevent against unauthorized access to any signing #
+// # functions.                                            #
+// #                                                       #
+// # Organizations that do not protect against un-         #
+// # authorized signing will be black-listed to prevent    #
+// # software piracy.                                      #
+// #                                                       #
+// # -QZ Industries, LLC                                   #
+// #                                                       #
+// #########################################################
+
+using System;
+using System.Text;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Security;
+using Org.BouncyCastle.Utilities.IO.Pem;
+
+namespace QZTray
+{
+    public class QZTraySigning
+    {
+
+        // Public method for signing the input string with the private key
+        // ===============================================================
+
+        // QZTray.key is a PEM-encoded private key of "QZTray.crt".
+        //
+        //     I.e. It's the key of the CSR (Certificate Signing Request), which is the
+        //          original certificate before being signed by the CA key.
+        private string _key = "QZTray.key";
+
+        public string SignMessage(string inputString)
+        {
+            // Convert the input string to a byte array
+            byte[] input = Encoding.ASCII.GetBytes(inputString);
+
+            // Initialize the signer with the algorithm and the private key
+            ISigner sig = SignerUtilities.GetSigner("SHA512withRSA");
+            sig.Init(true, getPrivateKey());
+
+            // Generate signature and return it as a base64 string
+            sig.BlockUpdate(input, 0, input.Length);
+            return Convert.ToBase64String(sig.GenerateSignature());
+        }
+
+        private AsymmetricKeyParameter getPrivateKey() {
+            using (var reader = System.IO.File.OpenText(_key))
+                return PrivateKeyFactory.CreateKey(new PemReader(reader).ReadPemObject().Content);
+        }
+
+
+        // Public method for returning the certificate
+        // ===========================================
+
+        // QZTray.crt is a PEM-encoded certificate signed by the CA (Certificate Authority).
+        //
+        //     Note: It is provided by "QZ Industries, LLC" as "digital-certificate.txt".
+        private string _cert = "QZTray.crt";
```

Comment:
Although there are advantages to using a controller for this, we don't provide this for our back-end examples due to the simplicity of just using AJAX/jquery/fetch.  For this reason, we won't accept this into our examples as it sets a precedent that we do not plan on continuing for all languages.